### ISSUE TITLE
The log-appender module is not working properly

### ISSUE THREAD

**Issue by qy-liuhuo** [open-telemetry/opentelemetry-java-examples#474]
The log-appender module does not function properly when using opentelemetry-collector-contrib:0.107.0, as evidenced by:

The docker container cannot receive log messages reported by the application.

I think the following container's logs are helpful in resolving the issue:

```
localhostgate/featuregate.go:63	The default endpoints for all servers in components have changed to use localhost instead of 0.0.0.0. Disable the feature gate to temporarily revert to the previous default.	{"feature gate ID": "component.UseLocalHostAsDefaultHost"}
```

** Problem fixed when I lowered the mirror version to 0.66.0 ** just change the `otel-config.yaml` as follow:

```
version: '3'
services:
  collector:
    image: otel/opentelemetry-collector-contrib:0.66.0
    volumes:
      - ./otel-config.yaml:/otel-config.yaml
    command: ["--config=/otel-config.yaml"]
    ports:
      - "4317:4317"
```

**Comment by qy-liuhuo** [comment]
I found a related issue in the opentelemetry-collector-contrib project：

[feature gate -The default endpoints for all servers in components have changed to use localhost instead of 0.0.0.0. #34394](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34394)

**Comment by jaydeluca** [comment]
fixed by #475
